### ISSUE TITLE
Bug - Strange encoding (40° => &#176)

### ISSUE THREAD

**Issue by johnykes** [nozzlegear/ShopifySharp#1118]
Hello,

I have some encoding problems and I don't know how to fix them.

For example when I try to set this title to a product:

shopifyProduct.Title= "Spiral flute 40°"

That's what I see in Shopify: `Spiral flute 40&#176`

I also have this problem with special characters specific to some language only.

How can I fix that?

Version: `6.19.0`

**Comment by nozzlegear** [comment]
Hey! My first suspicion is that this is something Shopify is doing on their end automatically, as that character sequence looks like the [HTML decimal reference](https://developer.mozilla.org/en-US/docs/Glossary/Character_reference#decimal_numeric_character_references) for the degree symbol. I'd bet if you looked at the product on the storefront, it would render as `Spiral flute 40°`.

Can you try changing the product title to another HTML character, just to confirm? 

| Input | Output |
| ----- | ------ |
| `Spiral flute 40 &` | `Spiral Flue 40 &#38` |
| `Spiral Flue 40 ®` | `Spiral Flue 40 &#174` |

**Comment by johnykes** [comment]
In shopify I see "&#176", not °.
If I manually edit in the admin dashboard to °, it works, but if I do the update using ShopifySharp, it's converted into &#176.
Couldn't find any setting/option to solve this.

**Comment by nozzlegear** [comment]
Ah okay, thanks for confirming! It sounds like it's a bug in ShopifySharp then. Are you using the `ProductService` class to update your product?

**Comment by johnykes** [comment]
Yes, I am using ProductService

**Comment by nozzlegear** [comment]
I'm thinking it's a bug in how we've configured the Json serializer. I'll add some unit tests and get this fixed!

**Comment by johnykes** [comment]
> I'm thinking it's a bug in how we've configured the Json serializer. I'll add some unit tests and get this fixed!

hello, any news on that?

**Comment by nozzlegear** [comment]
@johnykes Yes, sorry! I've got that pull request merged in that I had talked about previously. I still need to write the unit tests for this bug, I'll prioritize getting that done this week.

**Comment by johnykes** [comment]
> @johnykes Yes, sorry! I've got that pull request merged in that I had talked about previously. I still need to write the unit tests for this bug, I'll prioritize getting that done this week.

So it was merged in the last release, or it will be only after the unit tests? :) Sorry for spam.

**Comment by nozzlegear** [comment]
Hey @johnykes, sorry I missed your last message. I've added unit tests for this issue and I'm not seeing it in the latest builds of ShopifySharp unless you're using the GraphService. For all other services, it should encode/decode `Spiral Flute 40º` => `Spiral Flute 40º`. 

Can you try the latest prerelease version of ShopifySharp on Nuget and let me know if you still see this issue? It's [version 6.22.0-b232 on Nuget](https://www.nuget.org/packages/ShopifySharp/6.22.0-b232). Also, are you using the GraphService at all/have you used it in the past? That may have been what introduced the bug.

---

Quick side note: [all public apps are required to switch over to using the GraphQL product API by February 1st, and all custom apps are required to switch by April 1st](https://shopify.dev/docs/apps/build/graphql/migrate/new-product-model). I'm working on publishing more thorough docs on how to use ShopifySharp's GraphService today!

**Comment by johnykes** [comment]
> Hey [@johnykes](https://github.com/johnykes), sorry I missed your last message. I've added unit tests for this issue and I'm not seeing it in the latest builds of ShopifySharp unless you're using the GraphService. For all other services, it should encode/decode `Spiral Flute 40º` => `Spiral Flute 40º`.
> 
> Can you try the latest prerelease version of ShopifySharp on Nuget and let me know if you still see this issue? It's [version 6.22.0-b232 on Nuget](https://www.nuget.org/packages/ShopifySharp/6.22.0-b232). Also, are you using the GraphService at all/have you used it in the past? That may have been what introduced the bug.
> 
> Quick side note: [all public apps are required to switch over to using the GraphQL product API by February 1st, and all custom apps are required to switch by April 1st](https://shopify.dev/docs/apps/build/graphql/migrate/new-product-model). I'm working on publishing more thorough docs on how to use ShopifySharp's GraphService today!


Hey @nozzlegear ,

Thank you for all your time and support!

Just tested some 6.22 version these days and it worked great!

Regarding your quick note, will this impact me as a ShopifySharp user? Or will ShopifySharp just update the library calls to use GraphQL, and the library users should not be affected?

Thank you again!

**Comment by nozzlegear** [comment]
@johnykes Glad to hear it's working!

It will impact you, unfortunately. Shopify's REST API calls aren't a one-to-one match with their GraphQL calls, so there's a bit of migration to do depending on how complex your code is. If you're only doing "simple" product operations like getting a product and updating it, it shouldn't be too much work.

[Here's the link to the documentation for the GraphService](https://github.com/nozzlegear/shopifysharp/wiki/graphql), it uses the GraphQL product API as an example. Let me know if you have trouble with anything in particular, or if you have questions about migrating your current code to the GraphService. I'm happy to help!

**Comment by johnykes** [comment]
> [@johnykes](https://github.com/johnykes) Glad to hear it's working!
> 
> It will impact you, unfortunately. Shopify's REST API calls aren't a one-to-one match with their GraphQL calls, so there's a bit of migration to do depending on how complex your code is. If you're only doing "simple" product operations like getting a product and updating it, it shouldn't be too much work.
> 
> [Here's the link to the documentation for the GraphService](https://github.com/nozzlegear/shopifysharp/wiki/graphql), it uses the GraphQL product API as an example. Let me know if you have trouble with anything in particular, or if you have questions about migrating your current code to the GraphService. I'm happy to help!

Oh, just finished an integration and now I have to update 🤦  :))

Okay, it's cool that you have GraphQL too into this library. Thanks for the docs, will take a look!
